secure/push-metrics/src/counters.rs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::atomic::{AtomicBool, AtomicI64, Ordering};
+
+/// BaseMetric is a struct used by `Counter` and `Gauge`
+struct BaseMetric {
```

Comment:
actually it still isn't clear to me why Gauge doesn't just use Counter internally. I am not seeing a clear advantage to having them separate. And now that I see this code, I would argue Base or SimpleCounter would be fine. Though my preference would be to just merge this and Counter.

Reply:
There are a couple of differences b/w Gauge and Counter which is why I have to separate them.

1. Gauge values don't have a default value (unlike Counters which have a default of zero). So need to add special logic for that.
2. When reporting metrics to prometheus, we need to specify the "metric type" as "gauge" or "counter". So we need to have separate logic for Gauge and Counter in the `get_push_metric_format` function